backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -646,10 +646,12 @@ public ReleaseClearingStateSummary getReleaseClearingStateSummary(Set<String> id
         return out;
     }
 
-    private Optional<ReleaseLink> createReleaseLink(Set<String> visitedIds, Map.Entry<String, ?> entry, String id, Release release, Map<String, Release> releaseMap, String parentId) {
-        if (visitedIds.add(id)) {
+    private Optional<ReleaseLink> createReleaseLink(Stack<String> visitedIds, Map.Entry<String, ?> entry, String id, Release release, Map<String, Release> releaseMap, String parentId) {
+        ReleaseLink releaseLink = null;
```

Comment:
A lot of arguments ;-). If createReleaseLink is always called with id==entry.getKey() it's maybe worth considering to remove id from the argument list..